### PR TITLE
test case for splice-subquery and single shard db

### PR DESCRIPTION
**Pull request by mchacki**
This just adds an additional working test.
No production code change.

Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7366/

### CODE REVIEW

File: tests/js/server/aql/aql-subquery.js [hunk 1]
```diff
@@ -458,6 +458,48 @@ function ahuacatlSubqueryTestSuite () {
...
+        const rules = statement.explain().plan.rules;
+        // Has one shard rule
+        assertTrue(rules.indexOf("cluster-one-shard") !== -1);
+        // Has one splice subquery rule
+        assertTrue(rules.indexOf("splice-subqueries") !== -1);
+        // Result is as expected
+        const result = statement.execute().toArray();
+        for (let i = 0; i < 100; ++i) {
+          assertEqual(result[i].x.foo, i);
+          assertTrue(result[i].subquery.length, 1);
```
Comment (jsteemann) [inline tests/js/server/aql/aql-subquery.js]: should this rather be an `assertEqual`?

File: tests/js/server/aql/aql-subquery.js [hunk 2]
```diff
@@ -458,6 +458,48 @@ function ahuacatlSubqueryTestSuite () {
       } finally {
         db._drop(colName);
       }
+    },
+
+    testOneShardDBAndSpliceSubquery: function () {
+      const dbName = "SingleShardDB";
```
Comment (jsteemann) [inline tests/js/server/aql/aql-subquery.js]: We will probably need to add an `if` here that checks for an enterprise version build.
The one shard optimizer rule will not be triggered in community builds.

Comment (goedderz) [reply tests/js/server/aql/aql-subquery.js]: Or move the test into the enterprise repo.

**Comment by mchacki** [comment]
Jenkins after review comments:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7370/

**Comment by mchacki** [comment]
Jenkins Blue after latest commit:
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/7371/
Requested changes addressed

**Comment by mchacki** [comment]
As this has not been merged thus far, here is new Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7936/

**Comment by mchacki** [comment]
Updated to Latest devel.
New Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8348/

**Comment by mchacki** [comment]
The latest Jenkins triggered an unrelated issue.

**Pull request merged**